docs/book/en/src/dev/rust/wasinn.md

```diff
@@ -0,0 +1,196 @@
+```
+
+If everything goes well, you should have the terminal output:
+
+```bash
+# massive rust compiling output above...
+Read graph XML, size in bytes: 143525
+Read graph weights, size in bytes: 13956476
+Loaded graph into wasi-nn with ID: 0
+Created wasi-nn execution context with ID: 0
+Read input tensor, size in bytes: 602112
+Executed graph inference
+   1.) [963](0.7113)pizza, pizza pie
+   2.) [762](0.0707)restaurant, eating house, eating place, eatery
+   3.) [909](0.0364)wok
+   4.) [926](0.0155)hot pot, hotpot
+   5.) [567](0.0153)frying pan, frypan, skillet
+```
+
+## Code walkthrough
+
+The [main.rs](https://github.com/second-state/WasmEdge-WASINN-examples/blob/master/mobilenet/rust/src/main.rs) will be started as the entry point. First, read the model description and weights into memory:
+
+```rust
+let args: Vec<String> = env::args().collect();
+let model_xml_name: &str = &args[1]; // read filename from commandline
+let model_bin_name: &str = &args[2];
+let tensor_name: &str = &args[3];
+
+let xml = fs::read_to_string(model_xml_name).unwrap();
+let weights = fs::read(model_bin_name).unwrap();
+```
+
+Now we can start our inference with WASI-NN:
+
+```rust
+// load model
+let graph = unsafe {
+        wasi_nn::load(
+            &[&xml.into_bytes(), &weights],
+            wasi_nn::GRAPH_ENCODING_OPENVINO,
+            wasi_nn::EXECUTION_TARGET_CPU,
+        )
+        .unwrap()
+    };
+// initialize the computation context
+let context = unsafe { wasi_nn::init_execution_context(graph).unwrap() };
+// initialize the input tensor
+let tensor_data = fs::read(tensor_name).unwrap();
+let tensor = wasi_nn::Tensor {
+  dimensions: &[1, 3, 224, 224],
+  r#type: wasi_nn::TENSOR_TYPE_F32,
+  data: &tensor_data,
+};
+// set_input
+unsafe {
+  wasi_nn::set_input(context, 0, tensor).unwrap();
+}
+// Execute the inference.
+unsafe {
+  wasi_nn::compute(context).unwrap();
+}
+// retrieve output
+let mut output_buffer = vec![0f32; 1001];
+unsafe {
+  wasi_nn::get_output(
+    context,
+    0,
+    &mut output_buffer[..] as *mut [f32] as *mut u8,
+    (output_buffer.len() * 4).try_into().unwrap(),
+  )
+  .unwrap();
+}
+```
+
+where `wasi_nn::GRAPH_ENCODING_OPENVINO` means using the OpenVION™ backend, and `wasi_nn::EXECUTION_TARGET_CPU` means running the computation on CPU.
+
+Finally, we sort the output and then print the top-5 classification result:
+
+```rust
+let results = sort_results(&output_buffer);
+for i in 0..5 {
+  println!(
+    "   {}.) [{}]({:.4}){}",
+    i + 1,
+    results[i].0,
+    results[i].1,
+    imagenet_classes::IMAGENET_CLASSES[results[i].0]
+  );
+}
+```
+
+## Reference
+
+The intoduction of WASI-NN can be referred to [this amazing blog](https://bytecodealliance.org/articles/using-wasi-nn-in-wasmtime) written by Andrew Brown. This demo is greatly adapted from another [demo](https://github.com/radu-matei/wasi-nn-guest).
```

Comment:
Also, I would like to have a workflow to execute this example, maybe just inside the `build-extensions`.

Reply:
The example CI is in the example repo. Do you think we should put it back here?